src/extraction/index.js

```diff
@@ -20,31 +22,46 @@ import recordEvent from 'commerce/telemetry/content';
  * return either a valid ExtractedProduct, or null if a valid product could not
  * be found.
  */
-const EXTRACTION_METHODS = [
-  extractProductWithFathom,
-  extractProductWithFallback,
-  extractProductWithOpenGraph,
-];
+const EXTRACTION_METHODS = {
+  fathom: extractProductWithFathom,
+  css_selectors: extractProductWithCSSSelectors,
+  open_graph: extractProductWithOpenGraph,
+};
 
 /**
  * Perform product extraction, trying each method from EXTRACTION_METHODS in
  * order until one of them returns a truthy result.
  * @return {ExtractedProduct|null}
  */
-function extractProduct() {
-  for (const extract of EXTRACTION_METHODS) {
+function extractProduct(isBackgroundUpdate) {
+  const baseExtra = {
```

Comment:
The amount of events in this method are making it difficult to follow, what if we had a more concise API?

```js
function extractProduct() {
  const attempt = new ExtractionAttempt();
  attempt.start();
  for (const extract of EXTRACTION_METHODS) {
    const extractedProduct = extract(window.document);
    if (extractedProduct) {
      attempt.succeed(methodName);
      return extractedProduct;
    }
  }

  attempt.fail();
  return null;
}
```

Reply:
That's a great idea; thank you!